eva/binder/statement_binder.py

```diff
@@ -0,0 +1,159 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import singledispatchmethod
+from eva.binder.statement_binder_context import StatementBinderContext
+from eva.binder.binder_utils import bind_table_info, create_video_metadata
+from eva.catalog.catalog_manager import CatalogManager
+from eva.expression.abstract_expression import AbstractExpression
+from eva.expression.function_expression import FunctionExpression
+from eva.expression.tuple_value_expression import TupleValueExpression
+from eva.parser.create_mat_view_statement import \
+    CreateMaterializedViewStatement
+from eva.parser.load_statement import LoadDataStatement
+from eva.parser.select_statement import SelectStatement
+
+from eva.parser.statement import AbstractStatement
+from eva.parser.table_ref import TableRef
+from eva.parser.types import FileFormatType
+from eva.utils.generic_utils import path_to_class
+from eva.utils.logging_manager import LoggingLevel, LoggingManager
+
+
+class StatementBinder:
+    def __init__(self, binder_context: StatementBinderContext):
+        self._binder_context = binder_context
+        self._catalog = CatalogManager()
+
+    @singledispatchmethod
+    def bind(self, node: AbstractStatement):
```

Comment:
Is this type hint correct? It can be other class like tableref.

Reply:
Fixed